wandb/integration/keras/callbacks/model_checkpoint.py

```diff
@@ -171,3 +181,10 @@ def _check_filepath(self) -> None:
                 "This ensures correct interpretation of the logged artifacts.",
                 repeat=False,
             )
+
+    def _patch_tf_keras_version(self) -> None:
```

Comment:
the name is a bit misleading as you're not patching anything... I'd do something like
self._is_old_tf_keras_version: Optional[bool] = None
in __init__
then
convert this one into a property called is_old_tf_keras_version that sets the private self._is_old_tf_keras_version on first invocation and then just returns it -- there's no need to recalculate it every time.

Reply:
caching was not fixed, I added it in [67756e1](https://github.com/wandb/wandb/pull/4432/commits/67756e112f2b8ea87399e18868b5a1bcc3a34ad5)